proposals/format.md

```diff
@@ -0,0 +1,320 @@
+
+### Customizable interpolated strings
+Developers are able to customize the behavior of interpolated strings with `FormattableString`. This contains the data
+which goes into an interpolated string: the format `string` and the arguments as an array. This though still has the 
+boxing and argument array allocation as well as the allocation for `FormattableString` (it's an `abstract class`). Hence
+it's of little use to applications which are allocation heavy in `string` formatting.
+
+To make interopolated string formatting efficient the language will recognize a new type: 
+`System.ValueFormattableString`. All interpolated strings will have a target type conversion to this type. This will 
+be implemented by translating the interpolated string into the call `ValueFormattableString.Create` exactly as is done
+for `FormattableString.Create` today. The language will support all `params` options described in this document when
+looking for the most suitable `ValueFormattableString.Create` method. 
+
+``` csharp
+readonly struct ValueFormattableString {
+    public static ValueFormattableString Create(Variant v) { ... } 
+    public static ValueFormattableString Create(string s) { ... } 
+    public static ValueFormattableString Create(string s, params ReadOnlySpan<Variant> collection) { ... } 
+}
+
+class ConsoleEx { 
+    static void Write(ValueFormattableString f) { ... }
+}
+
+class Program { 
+    static void Main() { 
+        ConsoleEx.Write(42);
+        ConsoleEx.Write($"hello {DateTime.UtcNow}");
+
+        // Translates into 
+        ConsoleEx.Write(ValueFormattableString.Create((Variant)42));
+        ConsoleEx.Write(ValueFormattableString.Create(
+            "hello {0}", 
+            VariantCollection.Create((Variant)DateTime.UtcNow));
+    }
+}
+```
+
+Overload resolution rules will be changed to prefer `ValueFormattableString` over `string` when the argument is an 
+interpolated string. This means it will be valuable to have overloads which differ only on `string` and 
+`ValueFormattableString`. Such an overload today with `FormattableString` is not valauble as the compiler will always
+prefer the `string` version (unless the developer uses an explicit cast). 
+
+## Open Issues
+
+### ValuableFormattableString breaking change
+The change to prefer `ValueFormattableString` during overload resolution over `string` is a breaking change. It is
+possible for a developer to have defined a type called `ValueFormattableString` today and use it in method overloads
+with `string`. This proposed change would cause the compiler to pick a different overload once this set of features
+was implemented. 
+
+The possibility of this seems reasonably low. The type would need the full name `System.ValueFormattableString` and it 
+would need to have `static` methods named `Create`. Given that developers are strongly discouraged from defining any
+type in the `System` namespace this break seems like a reasonable compromise.
```

Comment:
It would be great if the Language/Compiler codified this (similer to how some languages carve out `__` names as being reserved for them).  If the language drew a line and said "we may change behavior for new types introduced in the `System...` space, and you should not be adding anything there yourself" then this gives both a reasonable path to add a lot of stuff in the future, while also being able to point users at some struct guidelines in case they do bad stuff.

Reply:
Leaving `System.` namespaces available for general use can allow third party libraries to implement features on older versions of the framework that aren't officially supported for that particular feature.  But having a compile warning may be appropriate.

Reply:
We've essentially drawn the line on types defined under `System.`. We're still pretty careful about introducing breaking changes here but if we feel we can improve things and it's a type under `System.` we'll accetp a level of breaking changes to do it.

Reply:
Great.  I wasn't certain if htat was something official, or more informal.  I was basically saying that it would be nice if really just stated it officially.